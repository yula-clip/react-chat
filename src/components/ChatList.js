import React from "react";
import { withStyles } from "@material-ui/core/styles";
import List from "@material-ui/core/List";
import ChatListItem from "./ChatListItem";
import Typography from "@material-ui/core/Typography";

const styles = theme => ({
  chatsList: {
    height: "calc(100% - 56px)",
    overflowY: "scroll"
  },
  noChats: {
    textAlign: "center"
  }
});

const ChatList = ({ classes, chats, activeChat, disabled }) => (
  <List className={classes.chatsList}>
    {chats && chats.length ? (
      chats.map(chat => (
        <ChatListItem
          disabled={disabled}
          key={chat._id}
          active={Boolean(activeChat && activeChat._id === chat._id)}
          chatId={chat._id}
          {...chat}
        />
      ))
    ) : (
      <Typography variant="subtitle1" className={classes.noChats}>
        There is no chats yet...
      </Typography>
    )}
  </List>
);

export default withStyles(styles)(ChatList);
