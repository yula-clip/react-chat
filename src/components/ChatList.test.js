/* eslint-env jest */
import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import ChatList from "./ChatList";

jest.mock("./ChatListItem", () => ({ chatId }) => `ChatListItem-${chatId}`);
jest.mock("moment", () => () => ({ fromNow: () => "2 days ago" }));

const mockProps = {
  chats: [
    {
      _id: "12345",
      title: "First chat",
      createdAt: "2018-03-16T10:53:23.200Z"
    },
    {
      _id: "54321",
      title: "Second chat",
      createdAt: "2018-03-16T10:53:23.200Z"
    }
  ],
  activeChat: {
    _id: "12345"
  },
  disabled: false
};

describe("<ChatList />", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ChatList {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<ChatList {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders without chats", () => {
    const tree = renderer
      .create(<ChatList {...mockProps} chats={[]} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
