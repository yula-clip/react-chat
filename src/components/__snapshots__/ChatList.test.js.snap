// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatList /> renders correctly 1`] = `
<ul
  className="MuiList-root-9 MuiList-padding-10 ChatList-chatsList-7"
>
  ChatListItem-12345
  ChatListItem-54321
</ul>
`;

exports[`<ChatList /> renders without chats 1`] = `
<ul
  className="MuiList-root-9 MuiList-padding-10 ChatList-chatsList-7"
>
  <h6
    className="MuiTypography-root-13 MuiTypography-subtitle1-31 ChatList-noChats-8"
  >
    There is no chats yet...
  </h6>
</ul>
`;
